goose.go

```diff
@@ -1051,136 +1065,142 @@ func (c *cursor) Remainder() []ast.Stmt {
 	return s
 }
 
-func endsWithReturn(s ast.Stmt) bool {
+func (ctx Ctx) endsWithReturn(s ast.Stmt) bool {
 	if s == nil {
 		return false
 	}
 	switch s := s.(type) {
 	case *ast.BlockStmt:
-		return stmtsEndWithReturn(s.List)
+		return ctx.stmtsEndWithReturn(s.List)
 	default:
-		return stmtsEndWithReturn([]ast.Stmt{s})
+		return ctx.stmtsEndWithReturn([]ast.Stmt{s})
 	}
 }
 
-func endIncludesIf(s ast.Stmt) bool {
-	if s == nil {
-		return false
-	}
-	if s, ok := s.(*ast.BlockStmt); ok {
-		return stmtsEndIncludesIf(s.List)
-	}
-	return stmtsEndIncludesIf([]ast.Stmt{s})
-}
-
-func stmtsEndWithReturn(ss []ast.Stmt) bool {
+func (ctx Ctx) stmtsEndWithReturn(ss []ast.Stmt) bool {
 	if len(ss) == 0 {
 		return false
 	}
-	// TODO: should also catch implicit continue
-	switch ss[len(ss)-1].(type) {
+	switch s := ss[len(ss)-1].(type) {
 	case *ast.ReturnStmt, *ast.BranchStmt:
 		return true
-	}
-	return false
-}
-
-func stmtsEndIncludesIf(ss []ast.Stmt) bool {
```

Comment:
I'm happy to remove these functions since I could not figure out the right contract for them anyway. ;)